System/Metrics.hs

```diff
@@ -355,101 +352,91 @@ sToMs s = round (s * 1000.0)
-     , ("rts.gc.max_bytes_used"           , Gauge . fromIntegral . Stats.max_live_bytes)
-     , ("rts.gc.current_bytes_used"       , Gauge . fromIntegral . Stats.gcdetails_live_bytes . Stats.gc)
-     , ("rts.gc.current_bytes_slop"       , Gauge . fromIntegral . Stats.gcdetails_slop_bytes . Stats.gc)
-     , ("rts.gc.max_bytes_slop"           , Gauge . fromIntegral . Stats.max_slop_bytes)
-     , ("rts.gc.peak_megabytes_allocated" , Gauge . fromIntegral . (`quot` (1024*1024)) . Stats.max_mem_in_use_bytes)
-     , ("rts.gc.par_tot_bytes_copied"     , Gauge . fromIntegral . Stats.par_copied_bytes)
-     , ("rts.gc.par_avg_bytes_copied"     , Gauge . fromIntegral . Stats.par_copied_bytes)
-     , ("rts.gc.par_max_bytes_copied"     , Gauge . fromIntegral . Stats.cumulative_par_max_copied_bytes)
+     , ("rts.mutator_cpu_ns"                       , Counter . Stats.mutator_cpu_ns)
+     , ("rts.mutator_elapsed_ns"                   , Counter . Stats.mutator_elapsed_ns)
+     , ("rts.gc_cpu_ns"                            , Counter . Stats.gc_cpu_ns)
+     , ("rts.gc_elapsed_ns"                        , Counter . Stats.gc_elapsed_ns)
+     , ("rts.cpu_ns"                               , Counter . Stats.cpu_ns)
+     , ("rts.elapsed_ns"                           , Counter . Stats.elapsed_ns)
+     -- GCDetails
+     , ("rts.gc.gen"                               , Gauge . fromIntegral . Stats.gcdetails_gen . Stats.gc)
+     , ("rts.gc.threads"                           , Gauge . fromIntegral . Stats.gcdetails_threads . Stats.gc)
+     , ("rts.gc.allocated_bytes"                   , Gauge . fromIntegral . Stats.gcdetails_allocated_bytes . Stats.gc)
+     , ("rts.gc.live_bytes"                        , Gauge . fromIntegral . Stats.gcdetails_live_bytes . Stats.gc)
+     , ("rts.gc.large_objects_bytes"               , Gauge . fromIntegral . Stats.gcdetails_large_objects_bytes . Stats.gc)
+     , ("rts.gc.compact_bytes"                     , Gauge . fromIntegral . Stats.gcdetails_compact_bytes . Stats.gc)
+     , ("rts.gc.slop_bytes"                        , Gauge . fromIntegral . Stats.gcdetails_slop_bytes . Stats.gc)
+     , ("rts.gc.mem_in_use_bytes"                  , Gauge . fromIntegral . Stats.gcdetails_mem_in_use_bytes . Stats.gc)
+     , ("rts.gc.copied_bytes"                      , Gauge . fromIntegral . Stats.gcdetails_copied_bytes . Stats.gc)
+     , ("rts.gc.par_max_copied_bytes"              , Gauge . fromIntegral . Stats.gcdetails_par_max_copied_bytes . Stats.gc)
+     , ("rts.gc.sync_elapsed_ns"                   , Gauge . fromIntegral . Stats.gcdetails_sync_elapsed_ns . Stats.gc)
+     , ("rts.gc.cpu_ns"                            , Gauge . fromIntegral . Stats.gcdetails_cpu_ns . Stats.gc)
+     , ("rts.gc.elapsed_ns"                        , Gauge . fromIntegral . Stats.gcdetails_elapsed_ns . Stats.gc)
      ])
     getRTSStats
 #else
+    -- Pre base-4.10 we have the names from before GHC commit 24e6594cc7890babe69b8ba122d171affabad2d1.
```

Comment:
Doesn't seem like a great idea to me. Can't we use new names with base-4.10 as well?

Reply:
Maybe, but then we'd have to put a new `ifdef` for that very version in which the old names were in place.

I didn't see the value in updating the EKG names for significantly older versions of `base`/`ghc`, but if you think there is one, I can try and do that.

Reply:
Our GHC support window is 3 years, otherwise I'd gladly drop support for `base < 4.10`.

Reply:
Done.